### PR TITLE
Fixed detection of console directory in symfony 3

### LINKED ISSUE #301: buggy implementation of `getConsoleDir` if `var/` is not existing
We just had the case that our deployment failed after upgrading to Symfony3 in the `clearCache` step (composer postinstall). 

The problem was that we didn't upload the `var/` directory to the container and therefore the check in `getConsoleDir -> useNewDirectoryStructure` tries to use `app/console` instead of `bin/console`. 

The implementation of `getConsoleDir` is less than ideal, IMHO. `getConsoleDir` should find a different check in which directory the console is. How about replacing that with this implementation?

```php

    /**
     * Returns a relative path to the directory that contains the `console` command.
     *
     * @param Event  $event      The command event
     * @param string $actionName The name of the action
     *
     * @return string|null The path to the console directory, null if not found.
     */
    protected static function getConsoleDir(Event $event, $actionName)
    {
        $options = static::getOptions($event);
        $binDirExists = static::hasDirectory($event, 'symfony-bin-dir', $options['symfony-bin-dir'], $actionName);
        
        if ($binDirExists && file_exists($binDirExists.DIRECTORY_SEPARATOR.'console')) {
            return $options['symfony-bin-dir'];
        }

        return $options['symfony-app-dir'];
    }
```

Related: #248

### PR DESCRIPTION
**Pull request by mweibel**
Closes #301.

### CODE REVIEW

File: Composer/ScriptHandler.php [hunk 1]
```diff
@@ -426,19 +426,12 @@ protected static function getPhpArguments()
...
     {
         $options = static::getOptions($event);
+        $binDirExists = isset($options['symfony-bin-dir']) && static::hasDirectory($event, 'symfony-bin-dir', $options['symfony-bin-dir'], $actionName);
 
-        if (static::useNewDirectoryStructure($options)) {
-            if (!static::hasDirectory($event, 'symfony-bin-dir', $options['symfony-bin-dir'], $actionName)) {
-                return;
-            }
-
+        if ($binDirExists && file_exists($binDirExists.DIRECTORY_SEPARATOR.'console')) {
```
Comment (HeahDude) [inline Composer/ScriptHandler.php]: This concatenation looks wrong, `$binDirExists` is a boolean not a path, am I missing something?

**Review by HeahDude — CHANGES_REQUESTED** [review]

File: Composer/ScriptHandler.php (hunk 1 above)
Comment (mweibel) [inline Composer/ScriptHandler.php]: d'oh ;) of course. Fixed.

**Review by HeahDude — APPROVED** [review]
👍

**Comment by HeahDude** [comment]
To be merged in 4.0.

**Comment by romain-pierre** [comment]
Why not returning the `bin` folder as default instead of the `app` one? So, making the same thing but testing first the presence of `app/console`. I'm experiencing the same issue as described in this case, but I think it's disturbing that the message said `Could not open input file: app/console` when I use Symfony 3.x and the console script has never been in the `app` folder.

Also, I don't understand why trying to run a script that we know absent from the folder. Maybe we can test the presence of this file instead of assuming (for a separated PR) and display a compliant error.

**Comment by fabpot** [comment]
Closing this pull request as this bundle is now deprecated. Symfony Flex replaces this bundle.

**Pull request closed without merging**